File: diquis-local-frontend/src/types/analytics.ts
export interface DailyCashflow {
    date: string;
    real_income: number;
    real_expense: number;
    projected_income: number;
    projected_expense: number;
    balance: number;
    is_future: boolean;
}

export interface CashflowReport {
    currency: string;
    start_date: string;
    end_date: string;
    starting_balance: number;
    ending_balance: number;
    timeline: DailyCashflow[];
}

File: diquis-local-backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List
from datetime import date

class DailyCashflow(BaseModel):
    date: date
    real_income: float
    real_expense: float
    projected_income: float
    projected_expense: float
    balance: float
    is_future: bool

class CashflowReport(BaseModel):
    currency: str 
    start_date: date
    end_date: date
    starting_balance: float
    ending_balance: float
    timeline: List[DailyCashflow]